### PR TITLE
fix(iOS): AutoSuggestBox Selection

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/11961

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?

Selection on an iOS AutoSuggestBox is not working. 

**Reason**: When selecting an option from the List, the `TextBox` sends an `EndEditing` notification that later ends in an `UnFocus()` event. This makes the `PopUp` associated with the `AutoSuggestBox` get dismissed before the PointerEvent is executed. 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
What happens when you tap somewhere outside the popup? I would guess in such case focus should be lost, or not?

**Comment by ajpinedam** [comment]
> What happens when you tap somewhere outside the popup? I would guess in such case focus should be lost, or not?

In that scenario, the focus works as expected, and the keyboard is dismissed.

https://github.com/unoplatform/uno/assets/1900897/5a27d721-9c68-4619-a38c-01c83f7005e1

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/AutoSuggestBox_SoftKeboard.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using Uno.UI.Samples.Controls;
+using Windows.UI.Xaml.Controls;
+
+// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
+
+namespace UITests.Windows_UI_Xaml_Controls.AutoSuggestBoxTests;
+
+[SampleControlInfo("AutoSuggestBox",
+				   "AutoSuggestBox_SoftKeboard",
+					description: "This test needs the Soft Keyboard",
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/AutoSuggestBox_SoftKeboard.xaml.cs:12]: Can you expand on the test expectations and steps?

**Review by jeromelaban — APPROVED** [review]

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/AutoSuggestBox_SoftKeboard.xaml.cs (hunk 1 above)
Comment (ajpinedam) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/AutoSuggestBox_SoftKeboard.xaml.cs:12]:
```suggestion
					description: "This test needs the Soft Keyboard. To validate the fix, start typing using the Keyboard on the Screen and tapping over one of the options should make the selection.",
```

Comment (ajpinedam) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/AutoSuggestBox_SoftKeboard.xaml.cs]: updated

**Pull request merged**